### ISSUE TITLE
CASE WHEN and IFNULL support

### ISSUE THREAD

**Issue by majcher** [sqlfluff/sqlfluff#894]
```
➜  sqlfluff cat test.sql
with table as (
  select 1 as column1, 2 as column2
  union all
  select 1 as column1, null as column2
  union all
  select 10 as column1, 2 as column2
  union all
  select 10 as column1, null as column2
)

select
  column1,
  column2,
  IF (((column1 - COALESCE (column2, 0)) > 5), "xx", "yy") AS column3,
  CASE WHEN ((column1 - COALESCE (column2, 0)) > 5) THEN "xx" ELSE "yy" END AS column4,
  CASE WHEN ((column1 - IFNULL (column2, 0)) > 5) THEN "xx" ELSE "yy" END AS column5,
from table
➜  sqlfluff sqlfluff parse --dialect bigquery test.sql
...Failed to Parse...
==== parsing violations ====
L:  14 | P:  58 |  PRS | Found unexpected end bracket!, was expecting one of: [<Ref:
                       | SemicolonSegment>], but got <class
                       | 'sqlfluff.core.parser.segments.raw.end_bracket_SymbolSegment'>

```

## Expected Behaviour
- `CASE WHEN` parses properly
- `IFNULL` parses properly - it's an alias for `COALESCE`

## Observed Behaviour
Parsing errors (see above)

## Steps to Reproduce
(see above)

## Version
Include the output of `sqlfluff --version` along with your Python version
```
➜  sqlfluff sqlfluff --version
sqlfluff, version 0.4.1
➜  sqlfluff python --version
Python 2.7.16
```

## Configuration
No config file, just CLI args.

**Comment by barrywhart** [comment]
The specific parse error mentioned in this issue ("Found unexpected end bracket") may have already been addressed in #836.

**Comment by barrywhart** [comment]
This query parses successfully in the latest master branch. The upcoming release (0.5.0, I think, but possibly 0.4.2) will contain this fix.

**Comment by majcher** [comment]
@barrywhart will you be able to have the 0.4.2 released anytime soon? The latest in pip atm is 0.4.1.

**Comment by barrywhart** [comment]
@majcher: That would be a question for @alanmcruickshank. I think it'll be released in the next two weeks or so.
